lib/manageiq/providers/vmware.rb

```diff
@@ -1 +1,21 @@
 require "manageiq/providers/vmware/engine"
+require "manageiq/providers/vmware/version"
+
+module ManageIQ
+  module Providers
+    module Vmware
+      def self.vix_disk_lib_installed?
+        return false unless RbConfig::CONFIG["host_os"].match?(/linux/i)
```

Comment:
Is this even necessary?

Reply:
I'm not sure, probably not?  The only thing I did was change it to use a built-in Ruby thing instead of requiring a gem.

Reply:
So, hilariously, it turns out this method doesn't even work at all.  Removing this line, and then running the method *always* returns true (I tried on Mac, and Linux-without-vix-disk-installed).

cc @agrare

Reply:
Ok, for now I'm going to leave this as is, and fix it in a follow up.  I think the "right" way to do this is for the gem itself to determine if it's installed, and then we just ask it.  So the code here would look something like:

```ruby
def self.vix_disk_lib_installed?
  require 'ffi-vix_disk_lib'
  FFI::VixDiskLib::API.available?
end
```

For the first part of this I made https://github.com/ManageIQ/ffi-vix_disk_lib/pull/14 , and I'll change this code later once that's vetted, merged, and released.

Reply:
Less is more!

Reply:
> So, hilariously, it turns out this method doesn't even work at all.

Do you mean that every platform is able to run the require regardless of the library being installed? Or do you mean the platform check doesn't work?

Reply:
Any platform can require the VixDiskLib.rb and it won't blow up (and least on the platforms I tried).  This linux check is essentially just preventing Macs.